DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs

```diff
@@ -0,0 +1,101 @@
+            return result.AccessToken;
+        }
+
+        private string doPUT(string URI, string body, String token, ILogger log)
+        {
+            Uri uri = new Uri(String.Format(URI));
+
+            // Create the request
+            var httpWebRequest = (HttpWebRequest) WebRequest.Create(uri);
+            httpWebRequest.Headers.Add(HttpRequestHeader.Authorization, "Bearer " + token);
+            httpWebRequest.ContentType = "application/json";
+            httpWebRequest.Method = "PUT";
+
+            try
+            {
+                using (var streamWriter = new StreamWriter(httpWebRequest.GetRequestStream()))
+                {
+                    streamWriter.Write(body);
+                    streamWriter.Flush();
+                    streamWriter.Close();
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogError("ex --> 1 " + ex.Message);
+            }
+
+            // Get the response
+            HttpWebResponse httpResponse = null;
+            string result = null;
+            try
+            {
+                httpResponse = (HttpWebResponse) httpWebRequest.GetResponse();
+            }
+            catch (Exception ex)
+            {
+                log.LogError("URI --> 2 " + URI);
+                log.LogError("body --> 2 " + body);
+                log.LogError("ex --> 2 " + ex.Message);
+                using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+                {
+                    result = streamReader.ReadToEnd();
+                    log.LogError("result --> 1 " + result);
+                }
+                return result;
+            }
+
+            using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+            {
+                result = streamReader.ReadToEnd();
+            }
+            return result;
+        }
+
+        [FunctionName("IncidentConsumer")]
+        public void Run([QueueTrigger("cohesity-incidents", Connection = "AzureWebJobsStorage")]string queueItem, ILogger log)
+        {
+            log.LogInformation("queueItem --> " + queueItem);
+            string token = GetAccessTokenAsync(ARM_ENDPOINT).Result;
+            log.LogInformation("token --> " + token);
+            string subscription = Environment.GetEnvironmentVariable("subscription");
+            string resourceGroup = Environment.GetEnvironmentVariable("resourceGroup");
+            string workspace = Environment.GetEnvironmentVariable("workspace");
+            dynamic body = JsonConvert.DeserializeObject(queueItem);
+            string incidentID = Guid.NewGuid().ToString();
+            string URI = $"{ARM_ENDPOINT}subscriptions/{subscription}/resourceGroups/{resourceGroup}/providers/Microsoft.OperationalInsights/workspaces/{workspace}/providers/Microsoft.SecurityInsights/incidents/{incidentID}?api-version=2021-10-01";
+            string result = doPUT(URI, body.ToString(), token, log);
```

Comment:
https://dev.to/kenakamu/azure-function-and-net-5-how-to-retry-with-queue-trigger-fik
By default, the function app retries five times before give-up processing the queue item. Then, it sends the item to Poison queue so that developer can take a look later.
so, my commit just throw new exception(), so the function would retry 5 times.

Reply:
I don't see the code that throws the exception. Where is it?

Reply:
you should see the code now.
add this code to set the retry count and interval.
        [FixedDelayRetry(5, "00:05:00")]